File: backend/meks/api/schemas/paper_analysis.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, field_serializer


class PaperAnalysisResponse(BaseModel):
    id: UUID
    document_id: UUID
    status: str
    data_statistics_score: int | None = None
    data_statistics_verdict: str | None = None
    logic_consistency_score: int | None = None
    logic_consistency_verdict: str | None = None
    credibility_score: int | None = None
    credibility_verdict: str | None = None
    overall_risk_score: int | None = None
    risk_level: str | None = None
    data_statistics_findings: str | None = None
    logic_consistency_findings: str | None = None
    credibility_findings: str | None = None
    reproducibility_score: int | None = None
    reproducibility_verdict: str | None = None
    reproducibility_findings: str | None = None
    figure_consistency_score: int | None = None
    figure_consistency_verdict: str | None = None
    figure_consistency_findings: str | None = None
    citation_manipulation_score: int | None = None
    citation_manipulation_verdict: str | None = None
    citation_manipulation_findings: str | None = None
    overall_summary: str | None = None
    recommendations: str | None = None
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

    @field_serializer("id", "document_id")
    def serialize_uuid(self, v: UUID) -> str:
        return str(v)

    @field_serializer("created_at", "updated_at")
    def serialize_dt(self, v: datetime) -> str:
        return v.isoformat()

File: frontend/src/api/paperAnalysis.ts
import apiClient from './client'

export interface PaperAnalysisResult {
  id: string
  document_id: string
  status: string
  data_statistics_score: number | null
  data_statistics_verdict: string | null
  logic_consistency_score: number | null
  logic_consistency_verdict: string | null
  credibility_score: number | null
  credibility_verdict: string | null
  overall_risk_score: number | null
  risk_level: string | null
  data_statistics_findings: string | null
  logic_consistency_findings: string | null
  credibility_findings: string | null
  reproducibility_score: number | null
  reproducibility_verdict: string | null
  reproducibility_findings: string | null
  figure_consistency_score: number | null
  figure_consistency_verdict: string | null
  figure_consistency_findings: string | null
  citation_manipulation_score: number | null
  citation_manipulation_verdict: string | null
  citation_manipulation_findings: string | null
  overall_summary: string | null
  recommendations: string | null
  error_message: string | null
  created_at: string
  updated_at: string
}

export interface PaperAnalysisTriggerResponse {
  detail: string
  document_id: string
}

export const paperAnalysisApi = {
  trigger: (documentId: string) =>
    apiClient.post<PaperAnalysisTriggerResponse>(`/paper-analysis/${documentId}`),
  get: (documentId: string) =>
    apiClient.get<PaperAnalysisResult>(`/paper-analysis/${documentId}`),
}
